docs/resources/Channel.md

```diff
@@ -246,6 +248,22 @@ Represents a message sent in a channel within Discord.
 | LISTEN | 3 |
 | JOIN_REQUEST | 5 |
 
+###### Message Flags
+
+| Type | Value | Description |
+| ---- | ----- | ----------- |
+| CROSSPOSTED | `0x00000001` | The message has been crossposted (published) to other channels |
+| IS\_CROSSPOST | `0x00000002` | The message is a crosspost of a message from another channel |
+| SUPPRESS\_EMBEDS | `0x00000004` | Do not include any embeds with this message |
+
+###### Message Reference
+
+| Field | Type | Description |
+| ----- | ---- | ----------- |
+| message\_id | snowflake | id of the referenced message |
+| channel\_id | snowflake | id of the channel where the referenced message was originally sent |
+| guild\_id? | snowflake | id of the guild where the referenced message was originally sent |
```

Comment:
We do not expect that `guild_id` will ever be null on a message reference going forward.

However, from the initial rollout of this feature, there are some message references that may not have a guild_id set. We may be able to update this to non-nullable in the future, but for now the nullability is added as a precaution.

Reply:
Just so ya know, if this should be null, we prepend the type with a `?`. Adding a `?` after the field name means optional

`guild_id?` = optional
`guild_id | ?snowflake` = nullable

Reply:
Yeah, so the field will not be sent if a `guild_id` isn't set. So if it is sent, it will be non-null, but it may not always be sent.